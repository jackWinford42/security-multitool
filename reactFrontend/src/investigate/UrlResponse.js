import "./Response.css";
import { Alert, Card, CardBody} from 'reactstrap';
import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined';

export default function UrlResponse({data, open}) {
  //console.debug("RESPONSE COMPONENT")
  const percent = 100 - data.risk_score;

  if (!data.success) return <Alert className="redAlert" color="danger">For best results, enter a valid url. Also, the url you entered may be unreachable.</Alert>
  return (
    <Card className="UrlResponse responseCard">
      <CardBody>
        <h5>This url is {percent}% safe <InfoOutlinedIcon id="infoIcon" onClick={() => open()}/></h5>
        <div className="progress">
          <div id="progressBar" className="progress-bar" role="progressbar" style={{width: `${percent}%`}} aria-valuenow={percent} aria-valuemin="0" aria-valuemax="100"></div>
        </div>
        <p>The type of content on this site falls into the category: {data.category}</p>
        {data.spamming && <p>The domain of this URL is associated with email SPAM or abusive email addresses.</p>}
        {data.malware && <p>This url is associated with malware and viruses.</p>}
        {data.phishing && <p>This url has been found to be associated with malicious phishing behavior.</p>}
        {data.dns_valid && <p>The domain of this URL has valid dns records.</p>}
      </CardBody>
    </Card>
  );
}
