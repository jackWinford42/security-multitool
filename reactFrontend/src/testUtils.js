import React from "react";
import { Provider, useDispatch } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers/rootReducer";
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)));

const demoUser = {
  email: "jackwinford@gmail.com",
  username: "captainjack",
  password: "qwerty",
  photo_url: "https://bit.ly/3C2VUgQ",
};

// const dispatch = useDispatch();
// await dispatch({type: "BEGIN_AUTH_SESSION", user: demoUser});

const UserProvider =(
    ({children}) =>
    (<Provider store={store}>
      {children}
    </Provider>)
);

export { UserProvider };