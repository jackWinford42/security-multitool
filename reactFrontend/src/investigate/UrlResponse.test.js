import React from "react";
import { render } from "@testing-library/react";
import UrlResponse from "./UrlResponse";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils"

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <UrlResponse data={{risk_score: 10, page_size: 1, spamming: false, malware: false, phishing: false, dns_valid: false}}/>
      </UserProvider>
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <UrlResponse data={{risk_score: 10, page_size: 1, spamming: false, malware: false, phishing: false, dns_valid: false}}/>
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});