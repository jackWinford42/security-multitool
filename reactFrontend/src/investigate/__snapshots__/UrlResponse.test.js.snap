// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="redAlert alert alert-danger fade"
    role="alert"
  >
    For best results, enter a valid url. Also, the url you entered may be unreachable.
  </div>
</DocumentFragment>
`;
